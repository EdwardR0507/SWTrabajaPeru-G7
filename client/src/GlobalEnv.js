module.exports = {
  //host: process.env.HOST || "http://localhost:4000",
  host: "https://ws-trabajaperu.herokuapp.com",
};
