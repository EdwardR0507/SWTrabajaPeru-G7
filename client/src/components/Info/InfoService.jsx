import { React, useState } from "react";
import { makeStyles, withStyles } from "@material-ui/core/styles";
import { Container, Button, Typography } from "@material-ui/core/";
import DeleteIcon from "@material-ui/icons/Delete";
import ServiceModal from "../Modals/ServiceModal";

const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    marginTop: "20px",
    marginBottom: "20px",
    width: "80%",
    height: "25vh",
    borderBottom: "1px solid #00000012",
  },
  description: {
    display: "flex",
    flexDirection: "column",
    width: "100%",
    textAlign: "justify",
  },
  button: {
    width: "9em",
    height: "2.8em",
    fontSize: "0.9em",
    lineHeight: "16px",
    letterSpacing: "1.25px",
  },
  image: {
    maxWidth: "100%",
    objectFit: "cover",
  },
}));

const StyledContainerData = withStyles({
  root: {
    width: "100%",
    display: "flex",
    alignItems: "center",
  },
})(Container);

const StyledContainerImage = withStyles({
  root: {
    width: "25%",
  },
})(Container);

const StyledContainerButtons = withStyles({
  root: {
    display: "flex",
    width: "70%",
    justifyContent: "space-evenly",
    alignItems: "center",
  },
})(Container);

const WrapContainer = withStyles({
  root: {
    width: "100%",
    padding: "0.1em",
    overflowWrap: "break-word",
  },
})(Container);

const InfoService = ({ cat_nombre, ser_descripcion }) => {
  const classes = useStyles();

  const [descriptionService, setDescriptionService] = useState(ser_descripcion);
  const [state, setState] = useState(true);

  const [modalDescription, setModalDescription] = useState(ser_descripcion);

  const handleDelete = () => {
    setState(false);
  };

  const handleEdit = (e) => {
    setModalDescription(e.target.value);
  };

  return (
    <>
      {state ? (
        <Container className={classes.root}>
          <StyledContainerData>
            {/*Aquí irá la imagen del servicio, primero importamos la imagen y luego la colocamos dentro del src, no olvidar poner el alt */}
            <StyledContainerImage>
              <></>
              {/*<img src={imageService} alt={"servicio"} className={classes.image} />*/}
            </StyledContainerImage>
            <Container className={classes.description}>
              <Typography color="primary" variant="subtitle1">
                {cat_nombre}
              </Typography>
              <WrapContainer>
                <Typography variant="body2">{descriptionService}</Typography>
              </WrapContainer>
            </Container>
          </StyledContainerData>
          <StyledContainerButtons>
            <ServiceModal
              serviceDescription={descriptionService}
              handleEdit={handleEdit}
              modalDescription={modalDescription}
              setDescriptionService={setDescriptionService}
            />
            <Button
              variant="contained"
              endIcon={<DeleteIcon />}
              className={classes.button}
              onClick={handleDelete}
            >
              Eliminar
            </Button>
          </StyledContainerButtons>
        </Container>
      ) : null}
    </>
  );
};

export default InfoService;
