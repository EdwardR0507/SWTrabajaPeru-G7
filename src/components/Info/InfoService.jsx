import { React, useState } from "react";
import {
  makeStyles,
  withStyles,
  Container,
  Typography,
} from "@material-ui/core/";
import ServiceModal from "../Modals/ServiceModal";
import DialogDelete from "../Dialog/DialogDelete";
import { useLocation } from "react-router";
import { fetchData } from "../../services/services";
import image from "../../assets/services.jpg";
const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    marginTop: "20px",
    marginBottom: "20px",
    width: "80%",
    height: "25vh",
    borderBottom: "1px solid #00000012",
  },
  description: {
    display: "flex",
    flexDirection: "column",
    width: "100%",
    textAlign: "justify",
  },
  image: {
    maxWidth: "100%",
    objectFit: "cover",
  },
}));

const StyledContainerData = withStyles({
  root: {
    width: "100%",
    display: "flex",
    alignItems: "center",
  },
})(Container);

const StyledContainerImage = withStyles({
  root: {
    width: "25%",
  },
})(Container);

const StyledContainerButtons = withStyles({
  root: {
    display: "flex",
    width: "70%",
    justifyContent: "space-evenly",
    alignItems: "center",
  },
})(Container);

const WrapContainer = withStyles({
  root: {
    width: "100%",
    padding: "0.1em",
    marginBottom: "2em",
    overflowWrap: "break-word",
    display: "inline-block",
    wordBreak: "break-word"
  },
})(Container);

const InfoService = ({ cat_id, cat_nombre, ser_descripcion }) => {
  const classes = useStyles();

  const [descriptionService, setDescriptionService] = useState(ser_descripcion);
  //Controlar el renderizado condicional, al ser false no se renderizará el componente InfoService
  const [stateRender, setStateRender] = useState(true);

  const [modalDescription, setModalDescription] = useState(ser_descripcion);

  const location = useLocation();
  const state = location.state;

  const handleEdit = (e) => {
    setModalDescription(e.target.value);
  };

  const handleDelete = () => {
    const newData = { cat_id: cat_id };
    fetchData(state?.token, "POST", "service-auth", "DELETE_SERVICE", newData);
    setStateRender(false);
  };

  return (
    <>
      {stateRender ? (
        <Container className={classes.root}>
          <StyledContainerData>
            {/*Aquí irá la imagen del servicio, primero importamos la imagen y luego la colocamos dentro del src, no olvidar poner el alt */}
            <StyledContainerImage>
              <img src={image} alt={"servicio"} className={classes.image} />
            </StyledContainerImage>
            <Container className={classes.description}>
              <Typography color="primary" variant="subtitle1">
                {cat_nombre}
              </Typography>
              <WrapContainer>
                <Typography variant="body2">{descriptionService}</Typography>
              </WrapContainer>
            </Container>
          </StyledContainerData>
          <StyledContainerButtons>
            <ServiceModal
              cat_nombre={cat_nombre}
              serviceDescription={descriptionService}
              role="edit"
              handleEdit={handleEdit}
              modalDescription={modalDescription}
              setDescriptionService={setDescriptionService}
            />
            <DialogDelete
              role="delete"
              cat_nombre={cat_nombre}
              handleDelete={handleDelete}
            />
          </StyledContainerButtons>
        </Container>
      ) : null}
    </>
  );
};

export default InfoService;
