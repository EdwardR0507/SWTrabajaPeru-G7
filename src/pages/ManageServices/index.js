import { React, useState, useEffect } from "react";
import { useLocation } from "react-router";
import NavBar from "../../layouts/NavBar";
import HeadingBar from "../../layouts/HeadingBar/HeadingBar";
import { Container, Typography, withStyles } from "@material-ui/core/";
import InfoService from "../../components/Info/InfoService.jsx";
import ServiceModal from "../../components/Modals/ServiceModal";
import { fetchData } from "../../services/services";

const StyledTypography = withStyles({
  root: {
    fontSize: "2.25rem",
    lineHeigth: "2.25em",
    fontStyle: "normal",
    fontWeigth: "400",
  },
})(Typography);

const StyledContainer = withStyles({
  root: {
    marginTop: "0.9em",
    paddingLeft: "9.4em",
    display: "flex",
    alignItems: "center",
  },
})(Container);

const ManageServices = () => {
  const location = useLocation();
  const state = location.state;

  const [user, setUser] = useState();
  const [data, setData] = useState([]);

  useEffect(() => {
    fetchData(state?.token, "GET", "user-auth", "GET_MY_USER")
      .then((res) => {
        setUser(res);
      })
      .catch((err) => {
        console.log(err);
      });
  }, [state?.token]);

  useEffect(() => {
    fetchData(state?.token, "GET", "service-auth", "GET_MY_SERVICES")
      .then((res) => {
        setData(res);
      })
      .catch((err) => {
        console.log(err);
      });
  }, [state?.token]);

  return user ? (
    <>
      <NavBar user={user} token={state?.token} />
      <HeadingBar before={"TRABAJADOR"} after={"MIS SERVICIOS"} />
      <StyledContainer>
        <StyledTypography>Mis Servicios</StyledTypography>
        <ServiceModal data={data} setData={setData} mood="Agregar" />
      </StyledContainer>
      {data.length !== 0 &&
        data.map((el) => {
          return (
            <InfoService
              key={`${el.cat_id}-${el.cat_nombre}-${el.ser_descripcion}`}
              {...el}
            />
          );
        })}
    </>
  ) : (
    <div>Cargando...</div>
  );
};
export default ManageServices;
