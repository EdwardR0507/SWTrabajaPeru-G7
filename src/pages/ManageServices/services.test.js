/*Importamos las librerias principales*/
import React from "react";
import { shallow } from "enzyme";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import GlobalEnv from "../../GlobalEnv";
import ManageServices from "./index";
import InfoService from "../../components/Info/InfoService";
/*Renderizado de la vista Social Profile*/
describe("<ManageServices></ManageServices>", () => {
  const server = setupServer(
    rest.post(`${GlobalEnv.host}/service-auth`, (req, res, ctx) => {
      return res(ctx.json({ greeting: "hello there" }));
    })
  );

  beforeAll(() => {
    server.listen();
  });
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => {
    server.close();
  });

  it("render `ManageServices`", () => {
    const history = createMemoryHistory();
    const wrapper = shallow(
      <Router history={history}>
        <ManageServices />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should not have an `InfoService` element", () => {
    const history = createMemoryHistory();
    const wrapper = shallow(
      <Router history={history}>
        <ManageServices />
      </Router>
    );
    expect(wrapper.contains(<InfoService />)).toBe(false);
  });
  it("show Cargando", () => {
    const history = createMemoryHistory();
    server.use(
      rest.post(`${GlobalEnv.host}/service-auth`, (req, res, ctx) => {
        return res(ctx.status(500));
      })
    );
  });
});
