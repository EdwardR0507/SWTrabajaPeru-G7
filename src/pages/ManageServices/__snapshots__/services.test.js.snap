// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageServices></ManageServices> render \`ManageServices\` 1`] = `ShallowWrapper {}`;
